### PR TITLE
Use CH session to store locales

### PR DESCRIPTION
**Pull request by HarryBosworth**
Change the LocaleResolver to use the Companies House session to resolve the locale.
This allows the locale to be passed easily over service boundaries. 
Also, adds the locale banner to change locale only when the locale is in session meaning it only shows for services that handle different locales.

- Add ChSessionLocaleResolver
- Update base layout to have locale banner

Resolves: [AOAF-523](https://companieshouse.atlassian.net/browse/AOAF-523)

[AOAF-523]: https://companieshouse.atlassian.net/browse/AOAF-523?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/lookup/internationalisation/SessionProvider.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package uk.gov.companieshouse.lookup.internationalisation;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.handler.SessionHandler;
+
+import java.util.Map;
+
+// Having an instance of this class allows to it to be mocked
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/lookup/internationalisation/SessionProvider.java:9]: I don't think you need this comment, this is a perfectly legitimate class

File: src/main/java/uk/gov/companieshouse/lookup/internationalisation/ChSessionLocaleResolver.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Locale;
+import java.util.Map;
+
+@Component
+public class ChSessionLocaleResolver implements LocaleResolver {
+    public static final String EXTRA_DATA_SESSION_KEY = "extra_data";
+    public static final String LANG_SESSION_KEY = "lang";
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/lookup/internationalisation/ChSessionLocaleResolver.java:17]: Should these be public?

File: src/main/java/uk/gov/companieshouse/lookup/internationalisation/SessionProvider.java (hunk 1 above)
Comment (HarryBosworth) [reply src/main/java/uk/gov/companieshouse/lookup/internationalisation/SessionProvider.java:9]: Removed

File: src/main/java/uk/gov/companieshouse/lookup/internationalisation/ChSessionLocaleResolver.java (hunk 2 above)
Comment (HarryBosworth) [reply src/main/java/uk/gov/companieshouse/lookup/internationalisation/ChSessionLocaleResolver.java:17]: I've made them private now

**Review by amartin7663 — APPROVED** [review]

**Pull request merged**